frontend-opal/src/components/PortfolioContainer/AllocationInfo/AllocationInfoContainer.js

```diff
@@ -0,0 +1,57 @@
+import { connect } from 'react-redux';
+import palette from 'google-palette';
+import getTotalInvestment from 'components/common/getTotalInvestment';
+import presenter from './AllocationInfoPresenter';
+
+const mapStateToProps = ({ portfolio }) => {
+  let sectors = {};
+  let totalInvestment = '';
+  if (portfolio[0]) {
+    totalInvestment = getTotalInvestment(portfolio);
+    sectors = portfolio.reduce((sectorsList, instrument) => {
+      const result = sectorsList;
+      if (result[instrument.sector]) {
+        result[instrument.sector].list.push(instrument);
+      } else {
+        result[instrument.sector] = { list: [instrument] };
+      }
+      return result;
+    }, {});
```

Comment:
@WenyuD let's extract this reduce into a separate function `groupBySectors(portfolio)`